app/dashboard/models.py

```diff
@@ -2658,6 +2658,25 @@ def activate_avatar(self, avatar_pk):
         self.avatar_baseavatar_related.filter(pk=avatar_pk).update(active=True)
 
     def to_dict(self, activities=True, leaderboards=True, network=None, tips=True):
+        from perftools.models import JSONStore
+        network = network or self.get_network()
+        key = f"profile_{self.pk}_{activities}_{leaderboards}_{network}_{tips}"
+        print(key)
+        return JSONStore.objects.get(key=key).data
+
+    def generate_to_dict(self, activities=True, leaderboards=True, network=None, tips=True):
+        from perftools.models import JSONStore
```

Comment:
I like the concept of storing the profile as a key-value hasmap kind of a data structure, but wouldn't it be advisable if we did it in redis instead of bloating the postgres database. My concern here is we have huge number of profile data, and if this is the approach we are taking, the database might bloat pretty fast. And also, since redis runs as a separate service, it will prevent us from any kind of blocking of the postgres service and interfere with other process. cc: @owocki @danlipert

Reply:
i believe that redis would certainly be faster but also cost more (it's a memory based data store, and memory is at premium).

i think the biggest issue here is the IOPS from reading from disk.  postgres is pretty good at storing JSONdata, but access times can be a problem for larger blobs IIRC.

there are also infrastructure dependancies here too.   i dont know that we have a good handle on the redis service in the docker and how much memory is allocated to it.

at the end of the day, idont feel super strongly here as either way would be an order of magnitude better than what were doing now (probably two orders of magnitude for big profiles like mine)